frontend/pages/datasets/index.vue

```diff
@@ -352,31 +367,33 @@ export default {
     onSearch(event) {
       this.querySearch = event;
     },
-    goToSetting(dataset) {
-      const { id, workspace, name, task } = dataset;
-      switch (task) {
-        case "TokenClassification":
-        case "TextClassification":
-        case "Text2Text":
-          this.$router.push({
-            path: `/datasets/${workspace}/${name}/settings`,
-          });
-          break;
-        case "FeedbackTask":
-          this.$router.push({
-            path: `/dataset/${id}/settings`,
-          });
-          break;
-        default:
-          console.log(`The task ${task} is unknown`);
+    goToSetting({ id, workspace, name, task }) {
+      const isOldTask = this.isOldTask(task);
+
+      if (isOldTask) {
+        this.$router.push({
+          path: `/datasets/${workspace}/${name}/settings`,
+        });
+      } else {
+        this.$router.push({
+          path: `/dataset/${id}/settings`,
```

Comment:
Could we use named routes here? This can help if we decide to change routes for datasets

See https://router.vuejs.org/guide/essentials/named-routes.html#named-routes